Devfile_Cert_Criteria.md

```diff
@@ -0,0 +1,84 @@
+# Certified Stack Hub
+
+This document describes the criteria a Devfile application stack must implement to be considered a certified application stack.
+
+## Certification Criteria
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in [BCP
+14](https://tools.ietf.org/html/bcp14)
+[[RFC2119](https://tools.ietf.org/html/rfc2119)]
+[[RFC8174](https://tools.ietf.org/html/rfc8174)] when, and only when, they
+appear in all capitals, as shown here.
+
+### Stack structure
+
+* A certified Application Stack MUST also be an [Application Stack](https://github.com/IBM/cp4apps-hub/tree/devfile_cert#application-stacks), having the necessary files needed to fulfill core scenarios by the architect (Champ) and the developer (Jane).
+
+* Stacks MUST have a deployment manifest defined in the devfile, using the `alpha.deployment-manifest` property ,that specifies the deployment of its application runtime container, using either:
+  * The Runtime Component Operator
+  * A runtime-specific Operator with equivalent functionality, such as the Open Liberty Operator
+
+* Stacks MUST contain one or more starter projects, which demonstrate best practices and programming models that a stack user can follow to get started.  
+
+### Capabilities
+
+* Stacks MUST provide mechanisms to execute "inner loop" functions used for developing, debugging and running (non-production) the microservice.  A common way is via a `Dockerfile`, but equivalent mechanisms are also acceptable, as long as they yield a development application container image.  
```

Comment:
Devfiles do not use cockerels for inner-loop

Reply:
Is it fair to say devfiles require you to provide a container image? Should this be worded something like....

"Stacks must provide mechanisms to execute "inner loop" functions used for developing, debugging and running (non-production) the microservice along with a development container image in which the actions are executed"?

Reply:
Yes, this seems correct. it is up to the stack creator to either provide a container image for dev. They might be able to use:

- pre-built publically available image
- custom build image which they maintain on their own
- pre-built image but use init containers (defined in devfile) to augment it with additional files.

But, we do not need to get to that level of detail in this doc.

Reply:
For " used for developing, debugging and running (non-production) ", add testing to the list.